addons/stock/models/stock_picking.py

```diff
@@ -850,7 +855,14 @@ def _compute_scheduled_date(self):
             else:
                 picking.scheduled_date = max(moves_dates, default=picking.scheduled_date or fields.Datetime.now())
 
-    @api.depends('move_line_ids', 'move_line_ids.result_package_id', 'move_line_ids.product_uom_id', 'move_line_ids.quantity')
+    @api.depends('move_ids.weight')
+    def _compute_weight(self):
+        done_pickings = self.filtered(lambda picking: picking.state == 'done')
+        not_done_pickings = self - done_pickings
```

Comment:
I think we shouldn't updated cancelled pickings too.
```suggestion
        pickings_to_update = self.filtered(lambda picking: picking.state not in ('done', 'cancel'))
```